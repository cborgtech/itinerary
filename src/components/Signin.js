import React, { Component } from "react";
import { firebaseApp } from "../firebase";
import history from "../history";
import { FormGroup, FormControl, InputGroup, Button } from "react-bootstrap";

class Signin extends Component {
  constructor(props, context) {
    super(props, context);

    this.handleChange = this.handleChange.bind(this);

    this.handleChangePass = this.handleChangePass.bind(this);

    this.state = {
      email: "",
      password: "",
      error: {
        message: ""
      }
    };
  }

  signIn() {
    const { email, password } = this.state;
    firebaseApp
      .auth()
      .signInWithEmailAndPassword(email, password)
      .catch(error => {
        this.setState({ error });
      });
  }

  historyUpdate() {
    history.replace("/signup");
  }

  getValidationState() {
    const str = this.state.email;
    const length = this.state.email.length;
    if (str.match(/@/) && str.indexOf(".com") > -1) return "success";
    else if (str.match(/@/) || str.indexOf(".com") > -1) return "warning";
    else if (length > 0) return "error";
    return null;
  }

  getValidationStatePass() {
    const passStr = this.state.password;
    const length = this.state.password.length;
    if (
      length > 10 &&
      passStr.match(/[~`!#$%^&*+=\-[\]\\;,/{}|\\":<>?]/) &&
      passStr.match(/[0-9]/)
    )
      return "success";
    else if (
      length > 10 ||
      passStr.match(/[~`!#$%^&*+=\-[\]\\;,/{}|\\":<>?]/) ||
      passStr.match(/[0-9]/)
    )
      return "warning";
    else if (length > 0) return "error";
    return null;
  }

  handleChange(e) {
    this.setState({ email: e.target.value });
  }

  handleChangePass(e) {
    this.setState({ password: e.target.value });
  }

  render() {
    return (
      <div style={{ margin: "5%" }}>
        <h1 className="title">Welcome to iTin3r@ry</h1>
        <form className="form-inline">
          <FormGroup
            controlId="emailInput"
            validationState={this.getValidationState()}
            style={{ marginRight: "5px" }}
          >
            <InputGroup>
              <InputGroup.Addon>@</InputGroup.Addon>
              <FormControl
                type="text"
                email={this.state.email}
                placeholder="Enter email"
                onChange={this.handleChange}
              />
            </InputGroup>
            <FormControl.Feedback />
          </FormGroup>
          <FormGroup
            controlId="passwordInput"
            validationState={this.getValidationStatePass()}
            style={{ marginRight: "5px" }}
          >
            <FormControl
              type="password"
              password={this.state.password}
              placeholder="Enter password (10 char, 1 sym and 1 num)"
              onChange={this.handleChangePass}
            />
            <FormControl.Feedback />
          </FormGroup>
          <Button bsStyle="success" onClick={() => this.signIn()}>
            Sign in to your Itinerary Account
          </Button>
        </form>
        <div
          className="error-msg"
          style={{ margin: "1%", paddingBottom: "6px" }}
        >
          {this.state.error.message}
        </div>
        <a
          className="signup-link"
          style={{ cursor: "pointer" }}
          onClick={() => this.historyUpdate()}
        >
          Not a member yet? Sign Up Here!
        </a>
      </div>
    );
  }
}

export default Signin;
