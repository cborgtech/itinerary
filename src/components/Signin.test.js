import React from "react";
import { shallow } from "enzyme";
import "../setupTest.js";
import Signin from "./Signin";

describe("The Sign-In Component", () => {
  const signin = shallow(<Signin />);
  const title = "Welcome to Itinerary";
  it("renders correctly", () => {
    expect(signin).toMatchSnapshot();
  });
  // it("displays a title", () => {
  //   expect(signin.find(".title").text()).toEqual(title);
  // });
});
