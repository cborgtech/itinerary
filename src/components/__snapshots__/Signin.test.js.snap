// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Sign-In Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signin />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
          Welcome to iTin3r@ry
</h1>,
        <form
          className="form-inline"
>
          <FormGroup
                    bsClass="form-group"
                    controlId="emailInput"
                    style={
                              Object {
                                        "marginRight": "5px",
                                      }
                    }
                    validationState={null}
          >
                    <InputGroup
                              bsClass="input-group"
                    >
                              <InputGroupAddon
                                        bsClass="input-group-addon"
                              >
                                        @
                              </InputGroupAddon>
                              <FormControl
                                        bsClass="form-control"
                                        componentClass="input"
                                        email=""
                                        onChange={[Function]}
                                        placeholder="Enter email"
                                        type="text"
                              />
                    </InputGroup>
                    <FormControlFeedback
                              bsClass="form-control-feedback"
                              bsRole="feedback"
                    />
          </FormGroup>
          <FormGroup
                    bsClass="form-group"
                    controlId="passwordInput"
                    style={
                              Object {
                                        "marginRight": "5px",
                                      }
                    }
                    validationState={null}
          >
                    <FormControl
                              bsClass="form-control"
                              componentClass="input"
                              onChange={[Function]}
                              password=""
                              placeholder="Enter password (10 char, 1 sym and 1 num)"
                              type="password"
                    />
                    <FormControlFeedback
                              bsClass="form-control-feedback"
                              bsRole="feedback"
                    />
          </FormGroup>
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="success"
                    disabled={false}
                    onClick={[Function]}
          >
                    Sign in to your Itinerary Account
          </Button>
</form>,
        <div
          className="error-msg"
          style={
                    Object {
                              "margin": "1%",
                            }
          }
/>,
        <a
          className="signup-link"
          onClick={[Function]}
          style={
                    Object {
                              "cursor": "pointer",
                            }
          }
>
          Not a member yet? Sign Up Here!
</a>,
      ],
      "style": Object {
        "margin": "5%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to iTin3r@ry",
          "className": "title",
        },
        "ref": null,
        "rendered": "Welcome to iTin3r@ry",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FormGroup
              bsClass="form-group"
              controlId="emailInput"
              style={
                            Object {
                                          "marginRight": "5px",
                                        }
              }
              validationState={null}
>
              <InputGroup
                            bsClass="input-group"
              >
                            <InputGroupAddon
                                          bsClass="input-group-addon"
                            >
                                          @
                            </InputGroupAddon>
                            <FormControl
                                          bsClass="form-control"
                                          componentClass="input"
                                          email=""
                                          onChange={[Function]}
                                          placeholder="Enter email"
                                          type="text"
                            />
              </InputGroup>
              <FormControlFeedback
                            bsClass="form-control-feedback"
                            bsRole="feedback"
              />
</FormGroup>,
            <FormGroup
              bsClass="form-group"
              controlId="passwordInput"
              style={
                            Object {
                                          "marginRight": "5px",
                                        }
              }
              validationState={null}
>
              <FormControl
                            bsClass="form-control"
                            componentClass="input"
                            onChange={[Function]}
                            password=""
                            placeholder="Enter password (10 char, 1 sym and 1 num)"
                            type="password"
              />
              <FormControlFeedback
                            bsClass="form-control-feedback"
                            bsRole="feedback"
              />
</FormGroup>,
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="success"
              disabled={false}
              onClick={[Function]}
>
              Sign in to your Itinerary Account
</Button>,
          ],
          "className": "form-inline",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <InputGroup
                  bsClass="input-group"
>
                  <InputGroupAddon
                                    bsClass="input-group-addon"
                  >
                                    @
                  </InputGroupAddon>
                  <FormControl
                                    bsClass="form-control"
                                    componentClass="input"
                                    email=""
                                    onChange={[Function]}
                                    placeholder="Enter email"
                                    type="text"
                  />
</InputGroup>,
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
/>,
              ],
              "controlId": "emailInput",
              "style": Object {
                "marginRight": "5px",
              },
              "validationState": null,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "input-group",
                  "children": Array [
                    <InputGroupAddon
                      bsClass="input-group-addon"
>
                      @
</InputGroupAddon>,
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      email=""
                      onChange={[Function]}
                      placeholder="Enter email"
                      type="text"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "input-group-addon",
                      "children": "@",
                    },
                    "ref": null,
                    "rendered": "@",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "form-control",
                      "componentClass": "input",
                      "email": "",
                      "onChange": [Function],
                      "placeholder": "Enter email",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control-feedback",
                  "bsRole": "feedback",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  password=""
                  placeholder="Enter password (10 char, 1 sym and 1 num)"
                  type="password"
/>,
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
/>,
              ],
              "controlId": "passwordInput",
              "style": Object {
                "marginRight": "5px",
              },
              "validationState": null,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "onChange": [Function],
                  "password": "",
                  "placeholder": "Enter password (10 char, 1 sym and 1 num)",
                  "type": "password",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control-feedback",
                  "bsRole": "feedback",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "success",
              "children": "Sign in to your Itinerary Account",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign in to your Itinerary Account",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "error-msg",
          "style": Object {
            "margin": "1%",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Not a member yet? Sign Up Here!",
          "className": "signup-link",
          "onClick": [Function],
          "style": Object {
            "cursor": "pointer",
          },
        },
        "ref": null,
        "rendered": "Not a member yet? Sign Up Here!",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
            Welcome to iTin3r@ry
</h1>,
          <form
            className="form-inline"
>
            <FormGroup
                        bsClass="form-group"
                        controlId="emailInput"
                        style={
                                    Object {
                                                "marginRight": "5px",
                                              }
                        }
                        validationState={null}
            >
                        <InputGroup
                                    bsClass="input-group"
                        >
                                    <InputGroupAddon
                                                bsClass="input-group-addon"
                                    >
                                                @
                                    </InputGroupAddon>
                                    <FormControl
                                                bsClass="form-control"
                                                componentClass="input"
                                                email=""
                                                onChange={[Function]}
                                                placeholder="Enter email"
                                                type="text"
                                    />
                        </InputGroup>
                        <FormControlFeedback
                                    bsClass="form-control-feedback"
                                    bsRole="feedback"
                        />
            </FormGroup>
            <FormGroup
                        bsClass="form-group"
                        controlId="passwordInput"
                        style={
                                    Object {
                                                "marginRight": "5px",
                                              }
                        }
                        validationState={null}
            >
                        <FormControl
                                    bsClass="form-control"
                                    componentClass="input"
                                    onChange={[Function]}
                                    password=""
                                    placeholder="Enter password (10 char, 1 sym and 1 num)"
                                    type="password"
                        />
                        <FormControlFeedback
                                    bsClass="form-control-feedback"
                                    bsRole="feedback"
                        />
            </FormGroup>
            <Button
                        active={false}
                        block={false}
                        bsClass="btn"
                        bsStyle="success"
                        disabled={false}
                        onClick={[Function]}
            >
                        Sign in to your Itinerary Account
            </Button>
</form>,
          <div
            className="error-msg"
            style={
                        Object {
                                    "margin": "1%",
                                  }
            }
/>,
          <a
            className="signup-link"
            onClick={[Function]}
            style={
                        Object {
                                    "cursor": "pointer",
                                  }
            }
>
            Not a member yet? Sign Up Here!
</a>,
        ],
        "style": Object {
          "margin": "5%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to iTin3r@ry",
            "className": "title",
          },
          "ref": null,
          "rendered": "Welcome to iTin3r@ry",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FormGroup
                bsClass="form-group"
                controlId="emailInput"
                style={
                                Object {
                                                "marginRight": "5px",
                                              }
                }
                validationState={null}
>
                <InputGroup
                                bsClass="input-group"
                >
                                <InputGroupAddon
                                                bsClass="input-group-addon"
                                >
                                                @
                                </InputGroupAddon>
                                <FormControl
                                                bsClass="form-control"
                                                componentClass="input"
                                                email=""
                                                onChange={[Function]}
                                                placeholder="Enter email"
                                                type="text"
                                />
                </InputGroup>
                <FormControlFeedback
                                bsClass="form-control-feedback"
                                bsRole="feedback"
                />
</FormGroup>,
              <FormGroup
                bsClass="form-group"
                controlId="passwordInput"
                style={
                                Object {
                                                "marginRight": "5px",
                                              }
                }
                validationState={null}
>
                <FormControl
                                bsClass="form-control"
                                componentClass="input"
                                onChange={[Function]}
                                password=""
                                placeholder="Enter password (10 char, 1 sym and 1 num)"
                                type="password"
                />
                <FormControlFeedback
                                bsClass="form-control-feedback"
                                bsRole="feedback"
                />
</FormGroup>,
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="success"
                disabled={false}
                onClick={[Function]}
>
                Sign in to your Itinerary Account
</Button>,
            ],
            "className": "form-inline",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <InputGroup
                    bsClass="input-group"
>
                    <InputGroupAddon
                                        bsClass="input-group-addon"
                    >
                                        @
                    </InputGroupAddon>
                    <FormControl
                                        bsClass="form-control"
                                        componentClass="input"
                                        email=""
                                        onChange={[Function]}
                                        placeholder="Enter email"
                                        type="text"
                    />
</InputGroup>,
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
/>,
                ],
                "controlId": "emailInput",
                "style": Object {
                  "marginRight": "5px",
                },
                "validationState": null,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "input-group",
                    "children": Array [
                      <InputGroupAddon
                        bsClass="input-group-addon"
>
                        @
</InputGroupAddon>,
                      <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        email=""
                        onChange={[Function]}
                        placeholder="Enter email"
                        type="text"
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "bsClass": "input-group-addon",
                        "children": "@",
                      },
                      "ref": null,
                      "rendered": "@",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "bsClass": "form-control",
                        "componentClass": "input",
                        "email": "",
                        "onChange": [Function],
                        "placeholder": "Enter email",
                        "type": "text",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control-feedback",
                    "bsRole": "feedback",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-group",
                "children": Array [
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    password=""
                    placeholder="Enter password (10 char, 1 sym and 1 num)"
                    type="password"
/>,
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
/>,
                ],
                "controlId": "passwordInput",
                "style": Object {
                  "marginRight": "5px",
                },
                "validationState": null,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control",
                    "componentClass": "input",
                    "onChange": [Function],
                    "password": "",
                    "placeholder": "Enter password (10 char, 1 sym and 1 num)",
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-control-feedback",
                    "bsRole": "feedback",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "success",
                "children": "Sign in to your Itinerary Account",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign in to your Itinerary Account",
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "error-msg",
            "style": Object {
              "margin": "1%",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Not a member yet? Sign Up Here!",
            "className": "signup-link",
            "onClick": [Function],
            "style": Object {
              "cursor": "pointer",
            },
          },
          "ref": null,
          "rendered": "Not a member yet? Sign Up Here!",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
